partial-update/src/main/java/com/azure/autorest/partialupdate/util/PartialUpdateHandler.java

```diff
@@ -122,23 +122,32 @@ private static String handlePartialUpdateForClassOrInterfaceFile(String generate
                 updatedMembersList.add(existingMember);
             } else {
                 // find the corresponding newly generated member
+                boolean hasFoundCorrespondingNewlyGeneratedMember = false;
                 for (BodyDeclaration<?> generatedMember : generatedFileMembers) {
                     if (isMembersCorresponding(existingMember, generatedMember)) {
+                        if (hasFoundCorrespondingNewlyGeneratedMember) {
+                            // it's not possible to find two methods in the generatedMembers with the same signature, otherwise generated file can have compilation error.
+                            throw new RuntimeException("Found more than one corresponding newly generated method. Generated file have methods with same signature and can ave compilation error");
+                        }
                         updatedMembersList.add(generatedMember);
+                        hasFoundCorrespondingNewlyGeneratedMember = true;
                     }
                 }
             }
         }
 
         // 5. add remaining members in generated file to the new members list
         for (BodyDeclaration<?> generatedMember : generatedFileMembers) {
-            boolean hasMembersWithSameName = false;
+            boolean needToAddToUpdateMembersList = true;
             for (BodyDeclaration<?> existingMember : updatedMembersList) {
-                if (isMembersWithSameName(existingMember, generatedMember)) {
-                    hasMembersWithSameName = true;
+                // If there is an existing member who has the same name as the generated member and is manually written,
+                // Or if the generated method and the existing method have the same method signature.
+                // Then we don't put the generated member to the updatedMembersList
+                if (isMembersWithSameName(existingMember, generatedMember) && !hasGeneratedAnnotation(existingMember) || isMembersCorresponding(existingMember, generatedMember)) {
```

Comment:
Move `isMembersCorresponding` as first condition (and maybe rename it to isMemebersSameSignature).
(as assume `isMembersWithSameName(existingMember, generatedMember) && !hasGeneratedAnnotation(existingMember)` is solely for method, and I am not sure whether this condition be to unpredictable for user)